### PR TITLE
docs: add intro and edit ignoring files page

### PR DESCRIPTION
**Pull request by bpmutter**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Add introduction and copy edits to Ignoring Files page

Partial fix of #16473

### CODE REVIEW

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (bpmutter) [inline docs/src/user-guide/configuring/ignoring-code.md:119]:
```suggestion
If an `.eslintignore` file is not found and an alternate file is not specified, ESLint looks in `package.json` for the `eslintIgnore` key to check for files to ignore.
```

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (mdjermanovic) [inline docs/src/user-guide/configuring/ignoring-code.md:12]: I'm not sure if it is correct to say "to ignore file patterns". Files are ignored, file patterns are a way to specify which files are ignored.

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (amareshsm) [inline docs/src/user-guide/configuring/ignoring-code.md:115]:
```suggestion
Any file that follows the standard ignore file format can be used. Keep in mind that specifying `--ignore-path` means that the existing `.eslintignore` file is not used. Note that globbing rules in `.eslintignore` follow those of `.gitignore`.
```

Only one `.eslintignore` file can be used at a time right?

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (nzakas) [reply docs/src/user-guide/configuring/ignoring-code.md:12]: Agree, maybe something like this?

```suggestion
You can configure ESLint to ignore certain files and directories while linting by specifying one or more glob patterns.
```

**Review by nzakas — CHANGES_REQUESTED** [review]
Looks good. I left a suggestion and there are a few others to look at.

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (snitin315) [reply docs/src/user-guide/configuring/ignoring-code.md:12]: This looks better 👍🏻

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (bpmutter) [reply docs/src/user-guide/configuring/ignoring-code.md:137]: 'an' is correct b/c '.eslintrc' starts w vowel sound.

**Comment by bpmutter** [comment]
applied suggestions from code review!

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (snitin315) [reply docs/src/user-guide/configuring/ignoring-code.md:137]: Depends on how we read it. I'm reading it as `Dot ESLint rc` not sure if that's correct.

Comment (nzakas) [reply docs/src/user-guide/configuring/ignoring-code.md:137]: I usually don’t say “dot”. I don’t think this matters too much so long as it’s consistent for both eslintrc and eslintignore.

**Review by nzakas — APPROVED** [review]
LGTM pending clarification of “a” vs “an” for eslintrc and eslintignore files.

**Review by amareshsm — APPROVED** [review]
LGTM

File: docs/src/user-guide/configuring/ignoring-code.md
Comment (snitin315) [reply docs/src/user-guide/configuring/ignoring-code.md:137]: In the same file above we have used "a", under the `## The .eslintignore File` section [here](https://github.com/eslint/eslint/pull/16510/files#diff-a336b2f4fd9fc95d4c173e4d02d9832abfbbec335c0ca3cf935865d9fd577e22R47) and [here too](https://github.com/eslint/eslint/pull/16510/files#diff-a336b2f4fd9fc95d4c173e4d02d9832abfbbec335c0ca3cf935865d9fd577e22R41).

Let's go with "a" for consistency.

Comment (bpmutter) [reply docs/src/user-guide/configuring/ignoring-code.md:137]: based on a quick search across the docs source files, we are actually evenly split with how we use (9 of each). 

will make a separate PR to be consistent w use, and will make that "an"

**Comment by bpmutter** [comment]
> LGTM pending clarification of “a” vs “an” for eslintrc and eslintignore files.

@nzakas see comment https://github.com/eslint/eslint/pull/16510#discussion_r1025900986 

i'll do the PR for 'an' consistency once all these docs changes are merged in.

**Review by harish-sethuraman — APPROVED** [review]
LGTM

**Review by snitin315 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**
